roles/kube_prometheus_stack/vars/main.yml

```diff
@@ -31,6 +31,24 @@ _kube_prometheus_stack_helm_values:
         openstack-control-plane: enabled
   grafana:
     adminPassword: "{{ kube_prometheus_stack_grafana_admin_password }}"
+    grafana.ini:
+      server:
+        root_url: https://{{ kube_prometheus_stack_grafana_host }}
+      auth.generic_oauth:
+        enabled: true
+        name: Keycloak-OAuth
+        allow_sign_up: true
+        client_id: "{{ atmosphere_grafana_oauth_client_id | default('grafana') }}"
+        client_secret: "{{ atmosphere_grafana_oauth_client_secret }}"
+        scopes: openid email profile offline_access roles
+        email_attribute_path: email
+        login_attribute_path: username
+        name_attribute_path: full_name
+        auth_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/auth"
+        token_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/token"
+        api_url: "https://{{ keycloak_host }}/realms/{{ atmosphere_keycloak_realm | default('Atmosphere') }}/protocol/openid-connect/userinfo"
+        tls_skip_verify_insecure: true
+        role_attribute_path: "contains(roles[*], 'admin') && 'Admin' || contains(roles[*], 'editor') && 'Editor' || 'Viewer'"
```

Comment:
@mnaser this role mapping is comming from grafana official doc. I don't have any other idea on this but have a hunch that we have to follow the Openstack role naming convention like member or reader. How about your idea?

Reply:
I think it's just a suggestion, I think what we need to do is the following:

- Create *application* roles inside Keycloak (so not global roles, roles that are application specific)
- User can assign users/groups access to Grafana with the admin/member/reader roles
- Grafana will not give Viewer permission, but 0 permissions if they do not have any of those roles.  This will stop any random cloud users of being able to access it.

Reply:
@mnaser  so it means we will create the roles like `grafana_admin`, `grafana_editor`, and `grafana_viewer` for Grafana, 
or
we will have `admin`, `member`, `reader` roles which present the perm levels and app-specific roles like `grafana`, etc.
Then we will check if the user has the set of roles like [grafana, admin], then grant grafana admin role, if the user has roles like `[grafana, viewer]` then grant viewer role. If he has role only `admin`,we will not grant grafana access.

Which one is perferrable?

Also are we gonna create those roles in the ansible playbook?